processor/metricstransformprocessor/metrics_testcase_builder_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+)
+
+type builder struct {
+	testcase *metricspb.Metric
+}
+
+// testcaseBuilder is used to build metrics testcases
+func testcaseBuilder() builder {
+	return builder{
+		testcase: &metricspb.Metric{
+			MetricDescriptor: &metricspb.MetricDescriptor{},
+			Timeseries:       make([]*metricspb.TimeSeries, 0),
+		},
+	}
+}
+
+// setName sets the name for the testcase
```

Comment:
You probably don't need many comments in this file, but if you do keep them I'd just say "metric" rather than "testcase". This code just builds metrics, they may or may not be used for testing. You should generally avoid comments that relate to how a consumer uses a function.

(having said that its not so bad in this case since these are unexported functions that can only be used by test code 😄)

```suggestion
// setName sets the name of the metric
```